backend/FwHeadless/Program.cs

```diff
@@ -146,10 +148,27 @@ static async Task<Results<Ok<ProjectSyncStatus>, NotFound>> GetMergeStatus(
         if (await projectLookupService.ProjectExists(projectId)) return TypedResults.Ok(ProjectSyncStatus.NeverSynced);
         else return TypedResults.NotFound();
     }
-    var commitsOnServer = await lexBoxDb.Set<ServerCommit>().CountAsync(c => c.ProjectId == projectId);
+    var lexboxProject = await lexBoxDb.Projects.Include(p => p.FlexProjectMetadata).FirstOrDefaultAsync(p => p.Id == projectId);
+    if (lexboxProject is null)
+    {
+        // Can't sync if lexbox doesn't have this project
+        return TypedResults.NotFound();
+    }
+    var crdtCommitsOnServer = await lexBoxDb.Set<ServerCommit>().CountAsync(c => c.ProjectId == projectId);
     var lcmCrdtDbContext = services.GetRequiredService<LcmCrdtDbContext>();
-    var localCommits = await lcmCrdtDbContext.Set<Commit>().CountAsync();
-    return TypedResults.Ok(ProjectSyncStatus.ReadyToSync(commitsOnServer - localCommits));
+    var localCrdtCommits = await lcmCrdtDbContext.Set<Commit>().CountAsync();
+    var pendingCrdtCommits = crdtCommitsOnServer - localCrdtCommits;
+    var lastCrdtCommitDate = await lcmCrdtDbContext.Set<Commit>().MaxAsync(commit => commit.DateTime);
+
+    var lastHgCommitDate = lexboxProject.LastCommit ?? DateTimeOffset.UnixEpoch;
+    var projectFolder = Path.Join(config.Value.ProjectStorageRoot, $"{lexboxProject.Code}-{projectId}");
+    if (!Directory.Exists(projectFolder)) Directory.CreateDirectory(projectFolder);
+    var fwDataProject = new FwDataProject("fw", projectFolder);
+    // TODO: Deicde what to return for pending Mercurial commits if we don't yet have a clone of the FW project. At the moment I'm going to return "-1" to mean "all of them, don't know how many".
+    var pendingHgCommits = File.Exists(fwDataProject.FilePath) ? await srService.PendingCommitCount(fwDataProject, lexboxProject.Code) : -1;
```

Comment:
yeah we can move this up. I would also move the code which checks if the file exists to inside the method to simplify the code here. 

I'm fine returning -1 here, just mention that on the ProjectSyncStatus so we don't forget to handle that case on the frontend when we implement that

Reply:
Done in commit f7cc7a8a for documenting the -1, and edb9e0cd for moving the `hg incoming` call up to the top so it gets kicked off first (and is awaited last, allowing the other tasks to complete in parallel).